### PR TITLE
Add product context handling to forms

### PR DESCRIPTION
**Pull request by caldav**
## Done

* Adds the notion of product context to modal forms and pass it to Marketo
* Modal contact forms now inherit the product parameter of the button they are triggered from
* If no parameter is set, the path of the page is passed instead, in the "pathlvl1-pathlvl2-etc" format.

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: [http://0.0.0.0:8001/](http://0.0.0.0:8001/)
- Click on any contact button, if it triggers a modal form, inspect and ensure it has a hidden "product-context" field set
- Navigate to `/ai`, click on the "Call the experts" button at the top page, inspect and ensure the form has its hidden "product-context" field set to `ai-index`
- Navigate to `/download/iot/raspberry-pi-2-3`, click on the "Contact us" button in the footer, inspect and ensure the form has its hidden "product-context" field set to `download-iot-raspberry-pi-2-3`

### CODE REVIEW

**Comment by anthonydillon** [comment]
To be clear, I thought you wanted multiple interactive forms to be triggered on the same page. This is just providing a hidden field with the path as context to the forms. Do we still want multiple interactive forms on the same page?

**Comment by caldav** [comment]
I think we still do yes, the page path is really a fallback for when there is no product parameter in the URL. Should I target a product-context class rather than an id then?

File: static/js/dynamic-contact-form.js [hunk 1]
```diff
@@ -17,6 +17,18 @@
...
+        // fallbacks to "global"
+        var product = window.location.pathname.split("/").slice(1).join("-") || "global";
+        // If present, override with product parameter from button URL
+        contactButton.search.split("&").forEach(function(param) {
+          if (param.startsWith("product") || param.startsWith("?product")) {
+            product = param.split("=")[1];
+          } 
+        });
+        // Set product in form field
+        document.getElementById("product-context").value = product;
```
Comment (anthonydillon) [inline static/js/dynamic-contact-form.js]: I think you should check the element is there before trying to set the value. Just to be more bullet proof.

**Comment by anthonydillon** [comment]
Some QA issues:
- If you visit http://0.0.0.0:8001/kubernetes/install#get-in-touch directly the value is None
- If you visit http://0.0.0.0:8001/internet-of-things/robotics and click the contact button the value is just robotics

**Comment by anthonydillon** [comment]
Updated the QA issues. Ready for a review by someone else.

File: static/js/dynamic-contact-form.js [hunk 2]
```diff
@@ -88,6 +89,29 @@
       });
     });
 
+    function setProductContext(contactButton) {
+      console.log('setProductContext');
```
Comment (steverydz) [inline static/js/dynamic-contact-form.js]: can we remove this console.log?

**Pull request merged**